### PR TITLE
Swage2hot

### PR DESCRIPTION
**Pull request by Atsumari**
#### Summary

Content "Added blacksmithing item"

#### Purpose of change

Makes swage and die into an item that is more prevalent to metalworking

#### Describe the solution

Replace swage with hotcut as a hotcut is used much more in metal working and makes more sense as to why it would be needed.

#### Describe alternatives you've considered

None.

#### Testing

spawned in all relative items to ensure they still worked.

#### Additional context

I am a blacksmith In my spare time and almost never use a swage block unless I'm mass producing items.

A hot cut I use every day as it literally just cuts the metal you're working on.

### CODE REVIEW

**Comment by I-am-Erk** [comment]
You're going to want to keep the item ID for swage, or people are going to have problems whenever they had a swage already in their game somewhere.

I suggest making hotcut a new item, and replacing swage with it as you've done, but keeping the definition to swage. We can make swages less common then, and include them occasionally in the rare recipes that could use them

**Review by kevingranade — CHANGES_REQUESTED** [review]
The swage is used in a lot of ammunition reloading recipes where a hotcut is nonsensical.
It's abstracted to be "a set of swages", which I'm aware is not how anything works, but using an anvil for these is even less sensical.
I'm not qualified to review the blacksmithing recipes at a glance, but at least all the ammunition recipes need to be switched back to a swage, and the recipe and spawns of the swage and die kit need to be restored.

**Comment by Atsumari** [comment]
I made an obsolete recipe that should auto convert them to the new id. its
included in the pr now.

**Comment by I-am-Erk** [comment]
> 
> I made an obsolete recipe that should auto convert them to the new id. its included in the pr now.

OK, but that's not what Kevin and I are asking for. We don't want swages to be obsoleted, there are some things we still want them for (ammo being a good example, and I'm not sure but I think nails and rivets too). It's OK for hotcut to replace it in most recipes and be more common, I am entirely sure you know more about that than me, but completely removing swages is not our goal here, definitely not without you making a stronger case for it.

**Comment by Atsumari** [comment]
> > I made an obsolete recipe that should auto convert them to the new id. its included in the pr now.
> 
> OK, but that's not what Kevin and I are asking for. We don't want swages to be obsoleted, there are some things we still want them for (ammo being a good example, and I'm not sure but I think nails and rivets too). It's OK for hotcut to replace it in most recipes and be more common, I am entirely sure you know more about that than me, but completely removing swages is not our goal here, definitely not without you making a stronger case for it.

I was editing my comment while you posted as I realized the intent.

**Comment by I-am-Erk** [comment]
> I was editing my comment while you posted as I realized the intent.

Ah, fair enough.

We can probably salvage this PR without too much work. I think the *majority* of recipes would prefer a hot cut, I've been reading about it since you posted this and I do think you're right. How exactly we go about that is up to you, but the basic steps are:
- add the hot cut, but it's not there to replace a swage, it's a new thing.
- put hot cut as a new item in any spawn lists that include a swage, and often reduce frequency of swage spawning
- review recipes that you've changed and revert a few of them to swage. This can probably be done mostly by file, like you can discard changes on all ammo files.
- IDK if you would ever need both in one recipe. I'm an utter poser here fully fueled by google and a bit of bourbon.

**Comment by Atsumari** [comment]
> > I was editing my comment while you posted as I realized the intent.
> 
> Ah, fair enough.
> 
> We can probably salvage this PR without too much work. I think the _majority_ of recipes would prefer a hot cut, I've been reading about it since you posted this and I do think you're right. How exactly we go about that is up to you, but the basic steps are:
> 
> * add the hot cut, but it's not there to replace a swage, it's a new thing. You can use copy-from to save much work if the two are pretty similar in dimensions and stuff.
> * put hot cut as a new item in any spawn lists that include a swage, and often reduce frequency of swage spawning
> * review recipes that you've changed and revert a few of them to swage. This can probably be done mostly by file, like you can discard changes on all ammo files.
> * IDK if you would ever need both in one recipe. I'm an utter poser here fully fueled by google and a bit of bourbon.

Theres very rare cases youd use both; but even then they just use industrial cutters anyways.

**Comment by Atsumari** [comment]
Ill see what I can manage tomorrow. Thanks again.

**Comment by Maleclypse** [comment]
Ooof you need to talk to someone in #dev-help because you've had some kind of commit go horribly awry.

**Comment by Atsumari** [comment]
Trying to figure it out. XD

**Comment by Atsumari** [comment]
Everything should be fixed and ready now.

Please let me know if there's anything else you want changed.

File: data/json/recipes/tools/tools_hand.json [hunk 1]
```diff
@@ -1090,29 +1090,16 @@
   },
   {
     "type": "recipe",
-    "activity_level": "MODERATE_EXERCISE",
+    "activity_level": "LIGHT_EXERCISE",
```
Comment (I-am-Erk) [inline data/json/recipes/tools/tools_hand.json:1093]: This looks like an error, you shouldn't be messing with washboard crafting in this pr

File: data/json/recipes/tools/tools_hand.json [hunk 2]
```diff
@@ -81,7 +81,7 @@
   {
     "type": "recipe",
     "activity_level": "BRISK_EXERCISE",
-    "result": "swage",
+    "result": "hotcut",
```
Comment (I-am-Erk) [inline data/json/recipes/tools/tools_hand.json:84]: I think you missed this one, you should have a recipe for both a swage and a hotcut, not replace the swage recipe with hotcut.

**Review by I-am-Erk — CHANGES_REQUESTED** [review]
In the home stretch now

**Comment by Atsumari** [comment]
> In the home stretch now

I'm assuming I'd just have both lines in each recipe then? And for the washboard I never changed that but I think it that edit was pushed by someone else and it pulled into my branch while I was working. I can change it back if you want.

**Comment by actual-nh** [comment]
> > In the home stretch now
> 
> I'm assuming I'd just have both lines in each recipe then? And for the washboard I never changed that but I think it that edit was pushed by someone else and it pulled into my branch while I was working. I can change it back if you want.

Current master has MODERATE_EXERCISE for the washboard so should be that after your PR. (And yes, there was a recent PR merged on that.)

Someone with permissions needs to run the JSON checker after you're done.

**Comment by actual-nh** [comment]
If you could lint the JSON files that are causing problems (click on the Details for JSON Validation to see which files; you can then also click on "Display Corrections" to get a better idea of the problem)?

**Pull request merged**